pkg/backend/diy/backend.go

```diff
@@ -795,29 +799,66 @@ func (b *diyBackend) ListStacks(
 		return nil, nil, err
 	}
 
+	// Get the max parallel value from environment variable or use a default value
+	maxParallel := b.Env.GetInt(env.DIYBackendParallel)
+	if maxParallel <= 0 {
+		maxParallel = 10 // Default to 10 parallel operations if not specified
+	}
+
 	// Note that the provided stack filter is only partially honored, since fields like organizations and tags
 	// aren't persisted in the diy backend.
-	results := slice.Prealloc[backend.StackSummary](len(stacks))
+	var filteredStacks []*diyBackendReference
 	for _, stackRef := range stacks {
 		// We can check for project name filter here, but be careful about legacy stores where project is always blank.
 		stackProject, hasProject := stackRef.Project()
 		if filter.Project != nil && hasProject && string(stackProject) != *filter.Project {
 			continue
 		}
+		filteredStacks = append(filteredStacks, stackRef)
+	}
 
-		chk, err := b.getCheckpoint(ctx, stackRef)
-		if err != nil {
-			// There is a race between listing stacks and getting their checkpoints.  If there's an error getting
-			// the checkpoint, check if the stack still exists before returning an error.
-			if _, existsErr := b.stackExists(ctx, stackRef); existsErr == errCheckpointNotFound {
-				continue
+	// Create a worker pool to process stacks in parallel
+	pool := newWorkerPool(maxParallel, len(filteredStacks))
+	defer pool.Close()
+	
+	// Create a slice to store results in the same order as filteredStacks
+	type checkpointResult struct {
+		ref *diyBackendReference
+		chk *apitype.CheckpointV3
+	}
+	results := make([]checkpointResult, len(filteredStacks))
+
+	// Enqueue work for each stack
+	for i, stackRef := range filteredStacks {
+		// No need to capture loop variables in Go 1.22+
+		pool.Enqueue(func() error {
+			chk, err := b.getCheckpoint(ctx, stackRef)
+			if err != nil {
+				// If checkpoint not found, stack doesn't exist anymore, don't report an error
```

Comment:
I think we need to fix b.getCheckpoint to return this error, currently it's only the exists code that does that.

I'm happy for you to either see about fixing and testing that, _or_ just put back the call to Exists but leave a TODO comment for us to re-look at it.